src/gui/application.cpp

```diff
@@ -661,60 +541,113 @@ static void displayHelpText(const QString &t, std::ostream &stream = std::cout)
 #endif
 }
 
-void Application::showHelp()
+void Application::parseOptions(const QStringList &arguments)
 {
-    setHelp();
-    QString helpText;
-    QTextStream stream(&helpText);
-    stream << _theme->appName()
-           << QLatin1String(" version ")
-           << _theme->version() << endl;
+    QCommandLineParser parser;
 
-    stream << QLatin1String("File synchronisation desktop utility.") << endl
-           << endl
-           << optionsC();
+    QString descriptionText;
+    QTextStream descriptionTextStream(&descriptionText);
+
+    descriptionTextStream << tr("%1 version %2\r\nFile synchronization desktop utility.").arg(_theme->appName()).arg(_theme->version()) << endl;
 
     if (_theme->appName() == QLatin1String("ownCloud")) {
-        stream << endl
-               << endl
-               << "For more information, see http://www.owncloud.org";
+        descriptionTextStream << endl
+                              << endl
+                              << tr("For more information, see http://www.owncloud.org");
     }
 
-    displayHelpText(helpText);
-}
+    parser.setApplicationDescription(descriptionText);
+
+    auto helpOption = parser.addHelpOption();
+    auto versionOption = parser.addVersionOption();
+
+    // this little snippet saves a few lines below
+    auto addOption = [&parser](const QCommandLineOption &option) {
+        parser.addOption(option);
+        return option;
+    };
+
+    auto showSettingsOption = addOption({ { "s", "showsettings" }, tr("Show the settings dialog while starting.") });
+    auto quitInstanceOption = addOption({ { "q", "quit" }, tr("Quit the running instance.") });
+    auto logFileOption = addOption({ "logfile", tr("Write log to file (use - to write to stdout)."), "filename" });
+    auto logDirOption = addOption({ "logdir", tr("Write each sync log output in a new file in folder."), "name" });
+    auto logExpireOption = addOption({ "logexpire", tr("Remove logs older than <hours> hours (to be used with --logdir)."), "hours" });
+    auto logFlushOption = addOption({ "logflush", tr("Flush the log file after every write.") });
+    auto logDebugOption = addOption({ "logdebug", tr("Output debug-level messages in the log.") });
+    auto languageOption = addOption({ "language", tr("Override UI language."), "language" });
+    auto listLanguagesOption = addOption({ "list-languages", tr("Override UI language.") });
+    auto confDirOption = addOption({ "confdir", tr("Use the given configuration folder."), "dirname" });
+    auto debugOption = addOption({ "debug", tr("Enable debug mode.") });
+
+    // virtual file system parameters (optional)
+    parser.addPositionalArgument("vfs file", tr("Virtual file system file to be opened (optional)."), { tr("[<vfs file>]") });
+
+    parser.process(*this);
```

Comment:
You need to conditionally check the passed arguments parameter.
This function is also invoked by ipc when a second instance is created.

Reply:
I'd recommend to not pass arguments in the default case but only when invoked from the ipc message.